From a29ef91fe09f5f4cdc38f9d3ac4514470b793682 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Sun, 04 Aug 2024 17:08:55 +0300
Subject: [PATCH] [FSW-7946] fs1: ppboot: onix: remove access to vuart

This patch is temporarily removing vuart from onix,
since this is not supported currently in Onix, and
result in getting stuck.

Change-Id: I477b3218bb35b7a45408613ec7267ad20722f9dc
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 0858ea0..338213e 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -197,7 +197,7 @@
 	/* not relevant in case of secure boot flow */
 }
 
-void preboot_putc_virt_uart(const char c)
+__weak void preboot_putc_virt_uart(const char c)
 {
 	regmap_writel(c, mmVUART_MNG0_TX_DATA);
 	regmap_writel(1, mmVUART_MNG0_TX_STATUS);
diff --git a/arch/fs1/arc/onix.c b/arch/fs1/arc/onix.c
index a32f744..954f7be 100644
--- a/arch/fs1/arc/onix.c
+++ b/arch/fs1/arc/onix.c
@@ -54,3 +54,6 @@
 
 /* PCI blocks no Available on Onix */
 void plat_pcie_drain_init(void) {}
+
+/* VUART not available in Onix */
+void preboot_putc_virt_uart(const char c) {}
